source/global/Data.gd

```diff
@@ -25,30 +25,92 @@ func _ready() -> void:
 
 
 func scan() -> void:
-	_load_terrain()
-	_load_races()
-	_load_units()
-	_load_factions()
-	_load_scenarios()
-	_load_ais()
+	terrains.clear()
+	units.clear()
+	races.clear()
+	factions.clear()
+	scenarios.clear()
+	AIs.clear()
+
+	_load_addons("user://addons/")
+
+	_load_terrain("res://data/terrain.gd")
+	_load_races("res://data/races")
+	_load_units("res://data/units")
+	_load_factions("res://data/factions")
+	_load_scenarios("res://data/scenarios")
+	_load_ais("res://ai")
+
+
+	var path = "res://addons/"
+
+	var directory := Directory.new()
+	directory.make_dir(path)
+
+	if not directory.open(path) == OK:
+		print("Loader: failed to load ", path, ", return [] (open)")
+		return
+
+	if not directory.list_dir_begin(true, true) == OK:
+		print("Loader: failed to load ", path, ", return [] (list_dir_begin)")
+		return
+
+	while true:
+		var sub_path = directory.get_next()
+		print(sub_path)
+
+		if sub_path == "." or sub_path == ".." or sub_path.begins_with("_"):
+			continue
+
+		elif sub_path == "":
+			break
```

Comment:
This seems to be the exact same code as in _load_addons(path) further down in the file. Why is this duplicated?